[Toolchain] Compilation errors with Host GCC 16.1.1

I am using ArchLinux. When following the tutorial [in the wiki](https://dreamcast.wiki/Getting_Started_with_Dreamcast_development#Setting_up_and_compiling_the_toolchain_with_the_dc-chain_script), I get these errors:
```
[/opt/toolchains/dreamcast/kos/utils/dc-chain] $ make
******************************************
Sega Dreamcast Toolchains Maker (dc-chain)
******************************************

+++ Using configuration from Makefile.cfg
...
...
make[3]: Entering directory '/opt/toolchains/dreamcast/kos/utils/dc-chain/build-gcc-sh-elf-13.2.0-pass1/libcody'
./config.status --header=config.h
config.status: creating config.h
config.status: config.h is unchanged
touch config.h
g++ -g -O2 -fno-exceptions -fno-rtti -include config.h -I../../gcc-13.2.0/libcody \
  -MMD -MP -MF buffer.d -c -o buffer.o ../../gcc-13.2.0/libcody/buffer.cc
In file included from ../../gcc-13.2.0/libcody/internal.hh:5,
                 from ../../gcc-13.2.0/libcody/buffer.cc:6:
../../gcc-13.2.0/libcody/cody.hh: In member function 'void Cody::Detail::MessageBuffer::Space()':
../../gcc-13.2.0/libcody/cody.hh:113:24: error: no matching function for call to 'S2C(const char8_t [2])'
  113 |     Append (Detail::S2C(u8" "));
      |             ~~~~~~~~~~~^~~~~~~
  * there is 1 candidate
    * candidate 1: 'template<unsigned int I> constexpr char Cody::Detail::S2C(const char (&)[I])'
      ../../gcc-13.2.0/libcody/cody.hh:51:16:
         51 | constexpr char S2C (char const (&s)[I])
            |                ^~~
      * template argument deduction/substitution failed:
        *   mismatched types 'const char' and 'const char8_t'
          ../../gcc-13.2.0/libcody/cody.hh:113:24:
            113 |     Append (Detail::S2C(u8" "));
                |             ~~~~~~~~~~~^~~~~~~
../../gcc-13.2.0/libcody/buffer.cc: At global scope:
../../gcc-13.2.0/libcody/buffer.cc:33:33: error: no matching function for call to 'S2C(const char8_t [2])'
   33 | static const char CONTINUE = S2C(u8";");
      |                              ~~~^~~~~~~
  * there is 1 candidate
    * candidate 1: 'template<unsigned int I> constexpr char Cody::Detail::S2C(const char (&)[I])'
      ../../gcc-13.2.0/libcody/cody.hh:51:16:
         51 | constexpr char S2C (char const (&s)[I])
            |                ^~~
      * template argument deduction/substitution failed:
        *   mismatched types 'const char' and 'const char8_t'
          ../../gcc-13.2.0/libcody/buffer.cc:33:33:
             33 | static const char CONTINUE = S2C(u8";");
                |                              ~~~^~~~~~~
../../gcc-13.2.0/libcody/buffer.cc: In member function 'void Cody::Detail::MessageBuffer::BeginLine()':
../../gcc-13.2.0/libcody/buffer.cc:41:28: error: no matching function for call to 'S2C(const char8_t [2])'
   41 |       buffer.push_back (S2C(u8" "));
      |                         ~~~^~~~~~~
  * there is 1 candidate
    * candidate 1: 'template<unsigned int I> constexpr char Cody::Detail::S2C(const char (&)[I])'
      ../../gcc-13.2.0/libcody/cody.hh:51:16:
         51 | constexpr char S2C (char const (&s)[I])
            |                ^~~
      * template argument deduction/substitution failed:
        *   mismatched types 'const char' and 'const char8_t'
          ../../gcc-13.2.0/libcody/buffer.cc:41:28:
             41 |       buffer.push_back (S2C(u8" "));
                |                         ~~~^~~~~~~

<More similar errors below>
```
Makefile.cfg is default with makejobs set to 1, Objective-C and Objective-C++ disabled (I commented the line for these 2).

Tested on both v2.2.x (commit c23410b3aa1ba6eeb00ac21a0634952edfaa1708) and master (commit 28fdd6527e6eaf703752d15925b3f0f55ab6ddc1) branch.
Compiling using GCC 15.3.0 from the package extra/gcc15 seems to fix the errors and I can run the process to completion.

--- a/utils/kos-chain/scripts/gcc-pass1.mk
+++ b/utils/kos-chain/scripts/gcc-pass1.mk
@@ -23,9 +23,8 @@ $(build_gcc_pass1): logdir
 	      $(gcc_pass1_configure_args) \
 	      $(macos_gcc_configure_args) \
 	      MAKEINFO=missing \
-	      CC="$(CC)" \
-	      CXX="$(CXX)" \
-	      CFLAGS="$(CFLAGS) -std=gnu17" \
+	      CC="$(CC) -std=gnu17" \
+	      CXX="$(CXX) -std=gnu++14" \
 	      $(static_flag) \
 	      $(to_log)
 	$(MAKE) $(jobs_arg) -C $(build) DESTDIR=$(DESTDIR) $(to_log)

--- a/utils/kos-chain/scripts/gcc-pass2.mk
+++ b/utils/kos-chain/scripts/gcc-pass2.mk
@@ -23,9 +23,8 @@ $(build_gcc_pass2): logdir
           $(gcc_pass2_configure_args) \
           $(macos_gcc_configure_args) \
           MAKEINFO=missing \
-          CC="$(CC)" \
-          CXX="$(CXX)" \
-          CFLAGS="$(CFLAGS) -std=gnu17" \
+          CC="$(CC) -std=gnu17" \
+          CXX="$(CXX) -std=gnu++14" \
           $(static_flag) \
           $(to_log)
 	$(MAKE) $(jobs_arg) -C $(build) DESTDIR=$(DESTDIR) $(to_log)

